src/utils/messages.ts

```diff
@@ -143,3 +148,42 @@ export const retrievedMessageToPublic = (
   sender_service_id: retrievedMessage.senderServiceId
 });
 /* eslint-enable @typescript-eslint/naming-convention */
+
+/**
+ * This function enrich a CreatedMessageWithoutContent with
+ * service's details and message's subject.
+ *
+ * @param messageModel
+ * @param serviceModel
+ * @param blobService
+ * @returns
+ */
+export const enrichMessageData = (
+  messageModel: MessageModel,
+  serviceModel: ServiceModel,
+  blobService: BlobService
+) => (
+  message: CreatedMessageWithoutContent
+): Promise<E.Either<Error, EnrichedMessage>> =>
+  pipe(
+    TE.Do,
+    TE.bind("service", () =>
+      serviceModel.findLastVersionByModelId([message.sender_service_id])
+    ),
+    TE.mapLeft(E.toError),
+    TE.bind("messageContent", () =>
+      messageModel.getContentFromBlob(blobService, message.id)
+    ),
+    TE.map(x => {
+      const content = O.getOrElse(() => ({} as MessageContent))(
+        x.messageContent
+      );
+      const service = O.getOrElse(() => ({} as Service))(x.service);
+      return {
+        ...message,
+        message_title: content.subject,
+        organization_name: service.organizationName,
+        service_name: service.serviceName
+      };
+    })
+  )();
```

Comment:
Not sure this function should be placed in a common repo, because its implementation could vary based on client needs. @balanza @BurnedMarshal ?

Reply:
I've choosen the place where there is also the implementation of the function that creates public messages. But I agree, this implementation should be in function app.

Reply:
I agree with @AleDore